service/src/main/java/tds/assessment/repositories/impl/AssessmentQueryRepositoryImpl.java

```diff
@@ -0,0 +1,88 @@
+package tds.assessment.repositories.impl;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.jdbc.core.namedparam.SqlParameterSource;
+import org.springframework.stereotype.Repository;
+import tds.assessment.Assessment;
+import tds.assessment.Segment;
+import tds.assessment.repositories.AssessmentQueryRepository;
+
+import javax.sql.DataSource;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+@Repository
+class AssessmentQueryRepositoryImpl implements AssessmentQueryRepository {
+    private static final Logger logger = LoggerFactory.getLogger(AssessmentQueryRepositoryImpl.class);
+    private final NamedParameterJdbcTemplate jdbcTemplate;
+
+    @Autowired
+    public AssessmentQueryRepositoryImpl(final DataSource dataSource) {
+        jdbcTemplate = new NamedParameterJdbcTemplate(dataSource);
+    }
+
+    @Override
+    public Optional<Assessment> findAssessmentByKey(final String assessmentKey) {
+        SqlParameterSource parameters = new MapSqlParameterSource("key", assessmentKey);
+
+        String SQL =
+            "SELECT \n" +
+                "A._key AS assessmentSegmentKey, \n" +
+                "testid AS assessmentSegmentId, \n" +
+                "selectionalgorithm AS selectionAlgorithm, \n" +
+                "startAbility, \n" +
+                "S.name, \n" +
+                "A.virtualtest AS assessmentKey\n" +
+            "FROM itembank.tblsetofadminsubjects A \n" +
+            "LEFT JOIN itembank.tblsubject S ON S._key = A._fk_Subject \n" +
+            "WHERE A.virtualtest = :key OR A._key = :key \n" +
+            "ORDER BY assessmentKey DESC";
+
+
+        List<Map<String,Object>> rows = jdbcTemplate.queryForList(SQL, parameters);
+        Assessment assessment = null;
```

Comment:
Why declare the Assessment here?  I'd probably have it in the `else` block since it is is only used in it.  Same for the Segments List creation.